tests/kw_ssh_test.sh

```diff
@@ -243,4 +243,96 @@ function test_ssh_remote()
   assertEquals "($LINENO)" "$expected_result" "$output"
 }
 
+function test_send_to_path_with_port_user_remote()
+{
+  local output
+  local expected_cmd
+
+  local -a inputs=(
+    'y' # overwrite it
+    'y' # overwrite it
+  )
+
+  touch 'some_file'
+  mkdir 'some_folder'
+
+  remote_parameters['REMOTE_FILE']=''
+  remote_parameters['REMOTE_FILE_HOST']=''
+  remote_parameters['REMOTE_USER']='mr_white'
+  remote_parameters['REMOTE_IP']='192.168.3.1'
+  remote_parameters['REMOTE_PORT']='3244'
+  options_values['SEND']="${PWD}/some_file"
+  options_values['TO']='/tmp'
+
+  output="$(printf '%s\n' "${inputs[@]}" | ssh_transfer_file "${options_values['SEND']}" 'send')"
```

Comment:
You are not using `flag` as a parameter inside `ssh_transfer_file`; in this sense, how is this test working? It should try to ssh somewhere, which should cause a failure.